lcdblib/odo_backends/rseqc.py

```diff
@@ -0,0 +1,115 @@
+"""Backends for shapeshifting RseQC output to pandas dataframes."""
+import re
+from odo import resource
+import pandas as pd
+
+
+class ParsingError(Exception):
+    pass
+
+
+@resource.register('.*infer_experiment\.txt', priority=20)
+def resource_infer_experiment(uri, **kwargs):
+    """odo resource to parse infer_experiment results.
+
+    Note
+    ----
+    The infer experiment file must be named `*infer_experiment.txt`.
+
+    """
+    header = [
+        'Fraction_reads_failed_assignment',
+        'Fraction_reads_on_gene_strand',
+        'Fraction_reads_opposite_gene_strand',
+        'Read_type',
+        ]
+
+    with open(uri, 'r') as fh:
+        data = fh.read().strip()
+        try:
+            # SingleEnd
+            se_regex = (
+                'This is SingleEnd Data\n'
+                'Fraction of reads failed to determine: (\d\.\d+)\n'
+                'Fraction of reads explained by \"\+\+\,\-\-\"\: (\d\.\d+)\n'
+                'Fraction of reads explained by \"\+\-\,\-\+\"\: (\d\.\d+)'
+                )
+
+            m = re.match(se_regex, data, flags=re.DOTALL).groups()
+            values = [float(x) for x in m]
+            values.append('SingleEnd')
+            return pd.Series(data=values, index=header)
+        except:
+            pass
+
+        try:
+            # PairEnd
+            pe_regex = (
+                'This is PairEnd Data\n'
+                'Fraction of reads failed to determine: (\d\.\d+)\n'
+                'Fraction of reads explained by \"1\+\+\,1\-\-,2\+\-\,2\-\+\"\: (\d\.\d+)\n'
+                'Fraction of reads explained by \"1\+\-\,1\-\+,2\+\+,2\-\-\"\: (\d\.\d+)'
+                )
+
+            m = re.match(pe_regex, data, flags=re.DOTALL).groups()
+            values = [float(x) for x in m]
+            values.append('PairEnd')
+            return pd.Series(data=values, index=header)
+        except:
+            pass
+
+        raise ParsingError('File did not conform to SE or PE read pattern.')
+
+@resource.register('.*geneBody_coverage\.txt', priority=20)
+def resource_geneBody_coverage(uri, **kwargs):
+    """odo resource to parse geneBody_coverage results.
+
+    Note
+    ----
+    The gene body coverage file must be named `*geneBody_coverage.txt`.
+
+    """
+    df = pd.read_csv(uri, sep='\t', index_col=0)
+    df.index = [x.replace('.prealn', '') for x in df.index]
```

Comment:
nevermind that is not true, it is workflow specific it, will figure out what needs change.d